server/src/test/java/org/opensearch/common/regex/RegexTests.java

```diff
@@ -96,6 +96,7 @@ public void testDoubleWildcardMatch() {
         assertFalse(Regex.simpleMatch("fff**ddd", "fffabcdd"));
         assertTrue(Regex.simpleMatch("fff*******ddd", "fffabcddd"));
         assertTrue(Regex.simpleMatch("fff*******ddd", "FffAbcdDd", true));
+        assertFalse(Regex.simpleMatch("fff*******ddd", "FffAbcdDd", false));
         assertFalse(Regex.simpleMatch("fff******ddd", "fffabcdd"));
```

Comment:
Could we add please more "bogus" tests:

```
        assertTrue(Regex.simpleMatch("*******************************************", "fffabcddd"));
        assertTrue(Regex.simpleMatch("************f*************b*******d***********", "fffabcddd"));
        assertTrue(Regex.simpleMatch("fff*******dd*****", "fffabcddd"));
        assertTrue(Regex.simpleMatch("****ff*******dd*****", "fffabcddd"));

```

Reply:
Yeah will do

Reply:
Need to make sure we cover all possible edge cases in the tests in both single and multiple wildcard cases
 - wildcard at start
 - wildcard at end
 - wildcard both at start and end
 - 0 wildcards
 - 1 wildcard
 - more than 1 wildcard